homeassistant/components/unifi/switch.py

```diff
@@ -284,3 +314,61 @@ async def options_updated(self) -> None:
         """Config entry options are updated, remove entity if option is disabled."""
         if self.client.mac not in self.controller.option_block_clients:
             await self.remove_item({self.client.mac})
+
+
+class UniFiDPIRestrictionSwitch(UniFiBase, SwitchEntity):
+    """Representation of a DPI restriction group."""
+
+    DOMAIN = DOMAIN
+    TYPE = DPI_SWITCH
+
+    @property
+    def key(self) -> Any:
+        """Return item key."""
+        return self._item.id
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this switch."""
+        return self._item.id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the client."""
+        return self._item.name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        if self._item.enabled:
+            return "mdi:network"
+        return "mdi:network-off"
+
+    @property
+    def is_on(self):
+        """Return true if client is allowed to connect."""
+        return self._item.enabled
+
+    async def async_turn_on(self, **kwargs):
+        """Turn on connectivity for client."""
+        await self.controller.api.dpi_groups.async_enable(self._item)
+
+    async def async_turn_off(self, **kwargs):
+        """Turn off connectivity for client."""
+        await self.controller.api.dpi_groups.async_disable(self._item)
+
+    async def options_updated(self) -> None:
+        """Config entry options are updated, remove entity if option is disabled."""
+        if not self.controller.option_dpi_restrictions:
+            await self.remove_item({self.key})
+
+    @property
+    def device_info(self) -> dict:
+        """Return a service description for device registry."""
+        return {
+            "identifiers": {(DOMAIN, f"{CONF_DPI_RESTRICTIONS}_{self._item.site_id}")},
+            "name": "Deep Packet Inspection Restrictions",
+            "manufacturer": ATTR_MANUFACTURER,
+            "model": "Deep Packet Inspection",
+            "entry_type": "service",
```

Comment:
Isn't this entity representing a feature of the unifi network switch? Then the device info should return info for the unifi network switch.

Reply:
It’s representing info from the controller, or second best the router. The problem is that we can’t really determine which router is on the device and updating the rules on the physical devices is done behind the scenes by the controller.

I could rename this service “UniFi Controller”. It’s still more of a service than a device, but then we can group all settings related configurations under this.

Reply:
Just to be clear, the device info should represent the thing the provides the entity feature. It should not represent the entity feature itself.

What is it that provides this entity feature? Are we saying that it's a service on the controller that provides this feature or can we just say that it's the controller that provides the feature?

Reply:
The controller is software, not a physical device. It can be installed as software on any machine, though newer equipment like the Dream Machine has the controller software installed on the routing device. There is also the "cloud key" which is older separate hardware that had the controller pre-installed. Therefore it's a service on the controller, not a physical device itself.

Reply:
Ok, good! Sounds good to name it UniFi Controller.